File: jawa/venator/src/main/java/ru/prohor/universe/venator/webhook/model/RepositoryInfo.java
package ru.prohor.universe.venator.webhook.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotNull;

public record RepositoryInfo(
        @NotNull @JsonProperty("full_name")
        String fullName,
        @NotNull @JsonProperty("pushed_at")
        long pushedAt,
        @NotNull @JsonProperty("clone_url")
        String cloneUrl,
        @NotNull @JsonProperty("master_branch")
        String masterBranch
) {}

File: jawa/venator/src/main/kotlin/webhook/model/RepositoryInfo.kt
package ru.prohor.universe.venator.webhook.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
data class RepositoryInfo(
    @field:JsonProperty("full_name")
    val fullName: String,
    @field:JsonProperty("pushed_at")
    val pushedAt: Long,
    @field:JsonProperty("master_branch")
    val masterBranch: String
)
